### ISSUE TITLE
how to generate embeddings for all entities after we have the model?

### ISSUE THREAD

**Issue by XRodriguez10** [facebookresearch/BLINK#21]
I'm trying to train a biencoder model to support Chinese. After I got the trained model for biencoder, how can I get the embeddings for all entities like the given file models/all_entities_large.t7?

**Comment by XRodriguez10** [comment]
Also, are you going to release instructions on how to train the models? Thanks!

**Comment by rajatinteros** [comment]
Hello All, I would also like to get some tips on training this architecture from scratch, also may use this architecture as a pre-train network on my custom dataset.

**Comment by anjalibhavan** [comment]
Yes, same issue here! I would like to know how to use this for a custom dataset, and how to generate embeddings from the linked documents.

**Comment by izuna385** [comment]
Hello, I just re-implemented hard-negative mining with zeshel dataset from [[Logeswaran et al., '19]](https://arxiv.org/abs/1906.07348).
See https://github.com/izuna385/Zero-Shot-Entity-Linking for your information.

**Comment by abhinavkulkarni** [comment]
Hi all,

You can refer to my comment https://github.com/facebookresearch/BLINK/issues/106#issuecomment-1014507351 with regards to generating embeddings for new candidates for an existing model.

**Comment by JLUGQQ** [comment]
> I'm trying to train a biencoder model to support Chinese. After I got the trained model for biencoder, how can I get the embeddings for all entities like the given file models/all_entities_large.t7?

I wonder whether you have trained this model using Chinese dataset. If so, can you share me your Chinese training dataset? I also want to use this model in Chinese, but I lack Chinese dataset. Thank you very much!

**Comment by abhinavkulkarni** [comment]
With regards to training a new model with custom data, yes, it is indeed possible to do so. I would recommend first training a zero-shot learning (zeshel) model first just to get hang of the training process. The scripts to download and pre-process zeshel data are in the repository. You can then replicate the same steps, bring your data in the same format as zeshel, modify any hyperparameters (such as context length or choice of bert base model) and train your own model.
